### PR TITLE
fix(forms-webapp): upload application and letter files are not mandat…

### PR DESCRIPTION
**Pull request by Fenisay**
UCD-899
UCD-884

### CODE REVIEW

**Comment by danWatkinson** [comment]
looks good; given the scrutiny we've had on this one i want to take the time to step over the test steps locally so i'm doing that.. in principal i approve, but i want to know what the video evidence is going to show me before i agree to merge

**Comment by Fenisay** [comment]
> looks good; given the scrutiny we've had on this one i want to take the time to step over the test steps locally so i'm doing that.. in principal i approve, but i want to know what the video evidence is going to show me before i agree to merge

It will be interesting to know as I'm not able to run e2e tests locally since yesterday and apparently @frank-dodion too

File: e2e-tests/cypress/integration/appellant-submission-decision-letter/appellant-submission-decision-letter.js [hunk 1]
```diff
@@ -1,45 +1,50 @@
...
 
-Given('user submits a decision letter file {string}', (filename) => {
+Given('user has previously submitted a decision letter file {string}', (filename) => {
   cy.goToDecisionLetterPage();
   cy.uploadDecisionLetterFile(filename);
+  cy.saveAndContinue();
 });
 
-Given('user does not submit a decision letter file', () => {
+When('user submits a decision letter file {string}', (filename) => {
```
Comment (frank-dodion) [inline e2e-tests/cypress/integration/appellant-submission-decision-letter/appellant-submission-decision-letter.js:9]: Is this identical to the one above?

**Review by frank-dodion — APPROVED** [review]
I see what looks like duplicated code in two steps but that can be addressed in refactoring later.

**Comment by danWatkinson** [comment]
@frank-dodion this is the exact test we just talked about for UCD-830, I'm thinking we need to take those additional features you added to 830 and implement them here.
right now, I believe this version of the test has the exact same problem 830 had when we spoke earlier: i'm not saying it doesn't work, just saying that the exact steps we're following wouldn't flag it if it broke

**Comment by frank-dodion** [comment]
> @frank-dodion this is the exact test we just talked about for [UCD-830](https://pins-ds.atlassian.net/browse/UCD-830), I'm thinking we need to take those additional features you added to 830 and implement them here.
> right now, I believe this version of the test has the exact same problem 830 had when we spoke earlier: i'm not saying it doesn't work, just saying that the exact steps we're following wouldn't flag it if it broke

OK. Let me share my UCD-830 approach with Yassine.

**Review by frank-dodion — APPROVED** [review]

**Review by danWatkinson — APPROVED** [review]

**Pull request merged**